space/space.go

```diff
@@ -302,7 +302,7 @@ func (r *GormRepository) LoadByOwner(ctx context.Context, userID *uuid.UUID, sta
 
 func (r *GormRepository) LoadByOwnerAndName(ctx context.Context, userID *uuid.UUID, spaceName *string) (*Space, error) {
 	res := Space{}
-	tx := r.db.Where("spaces.owner_id=? AND spaces.name=?", *userID, *spaceName).First(&res)
+	tx := r.db.Where("spaces.owner_id=? AND LOWER(spaces.name)=?", *userID, strings.ToLower(*spaceName)).First(&res)
```

Comment:
@kwk As you can see, the two queries use the index

```
postgres=# EXPLAIN SELECT * FROM spaces WHERE owner_id='00000000-0000-0000-0000-000000000000' AND name = 'test.space.yo' and deleted_at is null;
                                  QUERY PLAN
-------------------------------------------------------------------------------
 Index Scan using spaces_name_idx on spaces  (cost=0.28..8.30 rows=1 width=91)
   Index Cond: (owner_id = '00000000-0000-0000-0000-000000000000'::uuid)
   Filter: (name = 'test.space.yo'::text)
(3 rows)

postgres=# EXPLAIN SELECT * FROM spaces WHERE owner_id='00000000-0000-0000-0000-000000000000' AND lower(name) = 'test.space.yo' and deleted_at is null;
                                                     QUERY PLAN
---------------------------------------------------------------------------------------------------------------------
 Index Scan using spaces_name_idx on spaces  (cost=0.28..8.30 rows=1 width=91)
   Index Cond: ((owner_id = '00000000-0000-0000-0000-000000000000'::uuid) AND (lower(name) = 'test.space.yo'::text))
(2 rows)
```

for one index `CREATE UNIQUE INDEX spaces_name_idx ON spaces (owner_id, lower(name)) WHERE deleted_at IS NULL;
`